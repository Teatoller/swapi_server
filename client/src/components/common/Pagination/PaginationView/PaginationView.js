import React from "react";
import PropTypes from "prop-types";

export const PaginationView = ({
  handlePreviousPageChange,
  handleNextPageChange,
  peoplePage,
}) => (
    <nav aria-label="Page navigation example ">
      <ul className="pagination position-relative justify-content-center mb-5">
        <li className="page-item">
          <button
            type="button"
            onClick={handlePreviousPageChange}
            className="page-link"
            aria-label="Previous"
          >
            <span aria-hidden="true">&laquo;</span>
          </button>
        </li>
        <li className="page-item">
          <button type="button" className="page-link">
            {peoplePage}
          </button>
        </li>
        <li className="page-item">
          <button
            type="button"
            onClick={handleNextPageChange}
            className="page-link"
            aria-label="Next"
          >
            <span aria-hidden="true">&raquo;</span>
          </button>
        </li>
      </ul>
    </nav>
  );
  PaginationView.propTypes = {
    handlePreviousPageChange: PropTypes.func.isRequired,
    handleNextPageChange: PropTypes.func.isRequired,
    peoplePage: PropTypes.string.isRequired,
  };
export default PaginationView;
