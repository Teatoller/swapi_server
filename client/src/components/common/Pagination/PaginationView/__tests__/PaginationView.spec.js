import React from "react";
import renderer from "react-test-renderer";
import PaginationView from "../PaginationView";

describe("PaginationView", () => {
  it("snapshot renders correctly", () => {
    const component = renderer.create(<PaginationView/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
