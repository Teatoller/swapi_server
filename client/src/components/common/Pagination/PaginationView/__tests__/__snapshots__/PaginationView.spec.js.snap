// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationView snapshot renders correctly 1`] = `
<nav
  aria-label="Page navigation example "
>
  <ul
    className="pagination position-relative justify-content-center mb-5"
  >
    <li
      className="page-item"
    >
      <button
        aria-label="Previous"
        className="page-link"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          «
        </span>
      </button>
    </li>
    <li
      className="page-item"
    >
      <button
        className="page-link"
        type="button"
      />
    </li>
    <li
      className="page-item"
    >
      <button
        aria-label="Next"
        className="page-link"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          »
        </span>
      </button>
    </li>
  </ul>
</nav>
`;
